### PR TITLE
Refactor finalize logic

### PR DESCRIPTION
**Pull request by tcharding**
Attempt to simplify the `finalize` logic by doing an early return any time a condition is not met (incl. code comment as to why).

If none of the return statements are hit then we are ready to finalize.

### CODE REVIEW

File: comit/src/network/comit.rs [hunk 1]
```diff
@@ -210,32 +210,34 @@ impl Comit {
...
+                && (remote_data.lightning_identity.is_none() || !state.lightning_identity_sent)
+            {
+                // Swap not yet finalized, Lightning identities not synced.
+                return;
+            }
+
+            if data.bitcoin_identity.is_some()
+                && (remote_data.bitcoin_identity.is_none() || !state.bitcoin_identity_sent)
+            {
+                // Swap not yet finalized, Bitcoin identities not synced.
```
Comment (thomaseizinger) [inline comit/src/network/comit.rs:230]: I really like the notion of `synced`! Been struggling with finding a nice terminology here but `sync` could work really well :)

"Before a swap can take place, the parties first use the announce protocol verify essential properties of the swap like amounts and expiry times. After that, they synchronize technical execution parameters like secret-hash and identities."

Comment (tcharding) [reply comit/src/network/comit.rs:230]: Cheers, yeah seems like a good term for what we are doing.

**Comment by tcharding** [comment]
No one is reviewing, I'm kicking this one in

bors r+

**Pull request merged**